---  
title: "scRNA-seq: QC Report"
author: "SysBioLab"
output: 
  html_document:
    toc: true
    toc_float: true
    df_print: paged
    fig_width: 12
---

<!-- Sergio AlÃ­as, 20230706 -->
<!-- Last modified 20230723 -->

<style type='text/css'>
    body .main-container {
        max-width: 90%;

    }

    .plot_real_size {
      overflow: scroll;
      max-height: 600px;
    }

    .plot_real_size img{
      max-width: none;
      max-height: none;
    }

    embed {
   
  }   
</style>

<style>
.dropdown-content {
  display: none;
}
Error in parse(text = x, srcfile = src) : <text>:4:1: unexpected '<'
.dropdown:hover .dropdown-content {
  display: block;
}
</style>

<style>
  .superbigimage{
      overflow-x:scroll;
      white-space: nowrap;
  }

  .superbigimage img{
     max-width: none;
  }


</style>

# `r name` of project `r experiment`

Report of quality of FASTQ files using FastQC and Cell Ranger

## FastQC metrics

```{r setup, echo = FALSE}
library(patchwork)
library(ggplot2)

metric_df <- read.table(file = metrics, sep = '\t', header = TRUE)
long_metric_df <- read.table(file = long_metrics, sep = '\t', header = FALSE)
cellranger_metric_df <- read.table(file = cellranger_metrics, sep = '\t', header = TRUE)
cellranger_long_metric_df <- read.table(file = cellranger_long_metrics, sep = '\t', header = FALSE)
```

<div class="superbigimage">
```{r initial_total_sequences, eval=TRUE, results='asis', echo=FALSE}

cat("\n\n### **Number of sequences**\n\n")

make_barplot(metric_df, "initial_total_sequences")

```
</div>

<div class="superbigimage">
```{r initial_read_max_length, eval=TRUE, results='asis', echo=FALSE}

cat("\n\n### **Read maximum length**\n\n")

make_barplot(metric_df, "initial_read_max_length")

```
</div>

<div class="superbigimage">
```{r initial_read_min_length, eval=TRUE, results='asis', echo=FALSE}

cat("\n\n### **Read minimum length**\n\n")

make_barplot(metric_df, "initial_read_min_length")

```
</div>

<div class="superbigimage">
```{r initial_gc, eval=TRUE, results='asis', echo=FALSE}

cat("\n\n### **GC content**\n\n")

make_barplot(metric_df, "initial_.gc")

```
</div>

<div class="superbigimage">
```{r quality_scores, eval=TRUE, results='asis', echo=FALSE}

cat("\n\n### **Quality scores for samples - lowest scores (min/10th decile) and weighted average**\n\n")

make_hplot(long_metric_df, 
           c("initial_mean_qual_per_base",
             "initial_min_qual_per_base_in_lower_quartile",
             "initial_min_qual_per_base_in_10th_decile", 
             "initial_weigthed_qual_per_sequence"))

```
</div>

<div class="superbigimage">
```{r initial_mean_indeterminations_per_base, eval=TRUE, results='asis', echo=FALSE}

cat("\n\n### **Mean indeterminations per base**\n\n")

make_barplot(metric_df, "initial_mean_indeterminations_per_base")

```
</div>

<div class="superbigimage">
```{r initial_weigthed_read_length, eval=TRUE, results='asis', echo=FALSE}

cat("\n\n### **Weigthed read length**\n\n")

make_barplot(metric_df, "initial_weigthed_read_length")

```
</div>

## Cell Ranger metrics

<div class="superbigimage">
```{r Estimated_Number_of_Cells, eval=TRUE, results='asis', echo=FALSE}

cat("\n\n### **Estimated Number of Cells**\n\n")

make_barplot(cellranger_metric_df, "Estimated_Number_of_Cells")

```
</div>

<div class="superbigimage">
```{r Mean_Reads_per_Cell, eval=TRUE, results='asis', echo=FALSE}

cat("\n\n### **Mean Reads per Cell**\n\n")

make_barplot(cellranger_metric_df, "Mean_Reads_per_Cell")

```
</div>

<div class="superbigimage">
```{r Median_Genes_per_Cell, eval=TRUE, results='asis', echo=FALSE}

cat("\n\n### **Median Genes per Cell**\n\n")

make_barplot(cellranger_metric_df, "Median_Genes_per_Cell")

```
</div>

<div class="superbigimage">
```{r Valid_Barcodes, eval=TRUE, results='asis', echo=FALSE}

cat("\n\n### **Valid Barcodes percentage**\n\n")

make_barplot(cellranger_metric_df, "Valid_Barcodes")

```
</div>

<div class="superbigimage">
```{r Sequencing_Saturation, eval=TRUE, results='asis', echo=FALSE}

cat("\n\n### **Sequencing Saturation**\n\n")

make_barplot(cellranger_metric_df, "Sequencing_Saturation")

```
</div>

<div class="superbigimage">
```{r q30_bases_metrics, eval=TRUE, results='asis', echo=FALSE}

cat("\n\n### **Q30 Bases metrics**\n\n")

make_hplot(cellranger_long_metric_df, 
           c("Q30_Bases_in_Barcode",
             "Q30_Bases_in_RNA_Read",
             "Q30_Bases_in_UMI"))

```
</div>

<div class="superbigimage">
```{r mapped_reads_metrics, eval=TRUE, results='asis', echo=FALSE}

cat("\n\n### **Mapped reads metrics**\n\n")

make_hplot(cellranger_long_metric_df, 
           c("Reads_Mapped_to_Genome",
             "Reads_Mapped_Confidently_to_Genome",
             "Reads_Mapped_Confidently_to_Intergenic_Regions",
             "Reads_Mapped_Confidently_to_Intronic_Regions",
             "Reads_Mapped_Confidently_to_Exonic_Regions",
             "Reads_Mapped_Confidently_to_Transcriptome",
             "Reads_Mapped_Antisense_to_Gene"))

```
</div>

<div class="superbigimage">
```{r Fraction_Reads_in_Cells, eval=TRUE, results='asis', echo=FALSE}

cat("\n\n### **Fraction Reads in Cells**\n\n")

make_barplot(cellranger_metric_df, "Fraction_Reads_in_Cells")

```
</div>

<div class="superbigimage">
```{r Total_Genes_Detected, eval=TRUE, results='asis', echo=FALSE}

cat("\n\n### **Total Genes Detected**\n\n")

make_barplot(cellranger_metric_df, "Total_Genes_Detected")

```
</div>

<div class="superbigimage">
```{r Median_UMI_Counts_per_Cell, eval=TRUE, results='asis', echo=FALSE}

cat("\n\n### **Median UMI Counts per Cell**\n\n")

make_barplot(cellranger_metric_df, "Median_UMI_Counts_per_Cell")

```
</div>