## Quality Control

<!-- Sergio Alías, 20230616 -->
<!-- Last modified 20230620 -->

```{r qc_setup, echo = FALSE}

main_folder <- Sys.getenv("PREPROC_RESULTS_FOLDER")
seu <- readRDS(file.path(main_folder,
                         name,
                         "preprocessing.R_0000",
                         paste0(experiment,
                                ".",
                                name,
                                ".seu.RDS")))

```

### Before filtering

```{r violin_before, echo = FALSE}

VlnPlot(seu,
        features = c('nFeature_scRNAseq', 'nCount_scRNAseq', 'percent.mt', 'percent.rb'),
        ncol = 4)
```

- **nFeature_scRNAseq**: number of expressed (detected) genes/features per cell
- **nCount_scRNAseq**: number of reads per cell
- **percent.mt**: percentage of mitochondrial counts per cell
- **percent.rb**: percentage of ribosomal counts per cell


### Metadata features correlation


#### Mito counts VS nº of reads

```{r scatter1, echo = FALSE}

FeatureScatter(seu, feature1 = "nCount_scRNAseq", feature2 = "percent.mt")
```

#### Nº of features VS nº of reads

```{r scatter2, echo = FALSE}

FeatureScatter(seu, feature1 = "nCount_scRNAseq", feature2 = "nFeature_scRNAseq")
```

#### Ribosomal counts VS nº of reads

```{r scatter3, echo = FALSE}

FeatureScatter(seu, feature1 = "nCount_scRNAseq", feature2 = "percent.rb")
```

#### Mito counts VS ribosomal counts

```{r scatter4, echo = FALSE}

FeatureScatter(seu, feature1 = "percent.rb", feature2 = "percent.mt")
```


### Filtering parameters

```{r qc_filter, echo = FALSE}

table(seu[['QC']]) # TODO use package for nice table

```

- **High_MT**: Cells with more than **`r percentmt`%** of mitochondrial counts
- **High_MT,Low_nFeature**: Cells with more than **`r percentmt`%** of mitochondrial counts AND less than **`r minqcfeats`** detected genes/features
- **Pass**: Cells that passed the QC (less than **`r percentmt`%** of mitochondrial counts)

### After filtering

```{r violin_after, echo = FALSE}

VlnPlot(subset(seu, subset = QC == 'Pass'),
        features = c('nFeature_scRNAseq', 'nCount_scRNAseq', 'percent.mt', 'percent.rb'),
        ncol = 4)
```

If the plots after filtering are not as good as expected, try changing the QC parameters at `config_daemon`